### PR TITLE
[lcms] Backfill compat for lcms::lcms

### PR DESCRIPTION
**Pull request by ras0219-msft**
Backfill a few issues unaddressed in https://github.com/microsoft/vcpkg/pull/19551:

1. Follow upstream naming since we're breaking the binary names -- remove `d` suffix in debug
2. Add back compat support for `lcms::lcms` since there's no need to break existing users
3. Remove deprecated helpers (vcpkg_configure_cmake)
4. Switch to `version` scheme

### CODE REVIEW

File: ports/lcms/usage [hunk 1]
```diff
@@ -0,0 +1,4 @@
+The package lcms:@TARGET_TRIPLET@ provides CMake targets:
```
Comment (BillyONeal) [inline ports/lcms/usage:1]: I thought we said we don't want `:@TARGET_TRIPLET@` in this output elsewhere?

File: ports/lcms/portfile.cmake [hunk 2]
```diff
@@ -15,15 +15,19 @@ vcpkg_from_github(
...
 vcpkg_fixup_pkgconfig()
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
+file(WRITE "${CURRENT_PACKAGES_DIR}/share/lcms/lcms-config.cmake" [[
+include(CMakeFindDependencyMacro)
+find_dependency(lcms2 CONFIG)
+include(${CMAKE_CURRENT_LIST_DIR}/lcms-targets.cmake)
+]])
 
 file(INSTALL "${SOURCE_PATH}/COPYING" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
+configure_file("${CMAKE_CURRENT_LIST_DIR}/usage" "${CURRENT_PACKAGES_DIR}/share/${PORT}/usage" @ONLY)
```
Comment (BillyONeal) [inline ports/lcms/portfile.cmake:33]:
```suggestion
file(INSTALL "${CMAKE_CURRENT_LIST_DIR}/usage" "${CURRENT_PACKAGES_DIR}/share/${PORT}")
```

File: ports/lcms/usage (hunk 1 above)
Comment (BillyONeal) [reply ports/lcms/usage:1]:
```suggestion
The package lcms provides CMake targets:
```

Comment (BillyONeal) [reply ports/lcms/usage:1]: I'm not sure exactly which PR where this was litigated but I note that including the target triplet here prevents the usages from being deduplicated ( as in https://github.com/microsoft/vcpkg-tool/pull/175 ) . I filed https://github.com/microsoft/vcpkg-tool/pull/185 to make sure the heuristic ones also don't include that.

File: ports/lcms/portfile.cmake [hunk 3]
```diff
@@ -15,15 +15,19 @@ vcpkg_from_github(
...
-    PREFER_NINJA
-)
+vcpkg_cmake_configure(SOURCE_PATH "${SOURCE_PATH}")
+vcpkg_cmake_install()
 
-vcpkg_install_cmake()
 vcpkg_copy_pdbs()
-vcpkg_fixup_cmake_targets()
+vcpkg_cmake_config_fixup(PACKAGE_NAME lcms2)
+vcpkg_cmake_config_fixup()
```
Comment (NancyLi1013) [inline ports/lcms/portfile.cmake:23]: Why does it call `vcpkg_cmake_config_fixup` twice here?

**Comment by c72578** [comment]
Concerning the removal of the `d` suffix in debug, please consider `ports/libraw/lcms2_debug_fix.patch`:
https://github.com/microsoft/vcpkg/blob/615547c1d5b363d3586d2012c4cd7d075a0516b8/ports/libraw/lcms2_debug_fix.patch#L25

File: ports/lcms/portfile.cmake (hunk 3 above)
Comment (BillyONeal) [reply ports/lcms/portfile.cmake]: Because the second one provides the name "lcms" rather than "lcms2". I added a comment to indicate this.

**Review by BillyONeal — APPROVED** [review]
I'm happy as long as @c72578 's comment is addressed.

File: ports/lcms/portfile.cmake (hunk 3 above)
Comment (NancyLi1013) [reply ports/lcms/portfile.cmake]: Thanks for your clarification @BillyONeal.

File: ports/lcms/portfile.cmake [hunk 4]
```diff
@@ -15,15 +15,19 @@ vcpkg_from_github(
...
 vcpkg_fixup_pkgconfig()
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
+file(WRITE "${CURRENT_PACKAGES_DIR}/share/lcms/lcms-config.cmake" [[
+include(CMakeFindDependencyMacro)
+find_dependency(lcms2 CONFIG)
+include(${CMAKE_CURRENT_LIST_DIR}/lcms-targets.cmake)
+]])
 
 file(INSTALL "${SOURCE_PATH}/COPYING" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
+file(INSTALL "${CMAKE_CURRENT_LIST_DIR}/usage" "${CURRENT_PACKAGES_DIR}/share/${PORT}")
```
Comment (NancyLi1013) [inline ports/lcms/portfile.cmake:33]:
```suggestion
file(INSTALL "${CMAKE_CURRENT_LIST_DIR}/usage" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}")
```

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Comment by ras0219-msft** [comment]
Because vcpkg builds each flavor separately, the results will still be correct even given the patch: CMake will only ever find a "release" version (which is actually the debug version when built in debug) and will use that regardless of the consumer's configuration.

**Review by NancyLi1013 — APPROVED** [review]

**Pull request merged**